datalad/support/annexrepo.py

```diff
@@ -3476,24 +3465,56 @@ def _save_post(self, message, status, partial_commit):
         # first do standard GitRepo business
         super(AnnexRepo, self)._save_post(
             message, status, partial_commit)
-        # check if we have to deal with adjusted branch
+        # then sync potential managed branches
+        self.localsync()
```

Comment:
The "is corresponding branch" conditional is now removed from `localsync`, so shouldn't `_save_post` do it to avoid unnecessary `sync` calls?

Reply:
Makes sense. Although I spent a day trying graps all the combinations, and I am still not done. So....we'll see....

Reply:
I will introduce a `managed_only=False` parameter that allows for triggering the test from the outside.

Reply:
Sounds good.  And it looks like being able to avoid `annex sync` is necessary to prevent merge conflict failures like the one now happening in test_rerun.  Here's a reduced script that will trigger such a conflict without datalad:

<details>
<summary>conflict demo</summary>

```sh
#!/bin/sh

set -eux

cd "$(mktemp -d --tmpdir gx-XXXXXXX)"
git init
git annex init

echo foo >foo
git annex add foo
git commit -mfoo

git annex unlock foo
echo more >>foo
git annex add foo
git commit -m"foo more"

git annex sync --no-pull --no-push --no-commit --no-resolvemerge

git reset --hard HEAD~1
git annex unlock foo
echo different more >>foo
git annex add foo
git commit -m"foo different more"

git annex sync --no-pull --no-push --no-commit --no-resolvemerge
```

</details>

Reply:
I pushed that change. Need to detach now till some time next week.